altos-rust/altos-core/src/task/control.rs

```diff
@@ -63,6 +69,13 @@ impl Priority {
     (0..NUM_PRIORITIES)
   }
 
+
+  /// Returns a box of the range of values of Priority minus the one passed in.
+  pub fn all_except<'a>(priority:Priority) -> Box<Iterator<Item=usize> + 'a> {
+    let all = Priority::all();
+    Box::new(all.filter(move |&x| x != priority as usize))
```

Comment:
I like the idea of making our own iterator type. Another option could be using vectors. I'm not sure what kind of overhead would be incurred if we were to do so though. At first impression it seems pretty bad.